File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;
class Solution {
    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {        
        List<int[]> list = new ArrayList<>();
        
        int target = -1;
        if(ext.equals("code")) target = 0;
        if(ext.equals("date")) target = 1;
        if(ext.equals("maximum")) target = 2;
        if(ext.equals("remain")) target = 3;
        
        for(int i=0; i<data.length; i++) {
            if(data[i][target] < val_ext) list.add(data[i]);
        }
        
        Collections.sort(list, (o1, o2) -> {
            if(sort_by.equals("code")) return o1[0] - o2[0];
            if(sort_by.equals("date")) return o1[1] - o2[1];
            if(sort_by.equals("maximum")) return o1[2] - o2[2];
            if(sort_by.equals("remain")) return o1[3] - o2[3];
            return o1[0] - o2[0];
        });
        
        int[][] answer = new int[list.size()][4];
        for(int i=0; i<answer.length; i++) {
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    a = []
    target_num = 0
    
    if ext == "code" :
        target_num = 0
    elif ext == "date" :
        target_num = 1
    elif ext == "maximum" :
        target_num = 2
    elif ext == "remain" :
        target_num = 3
        
    for arr in data :
        if arr[target_num] < val_ext :
            a.append(arr)
    
    if sort_by == "code" :
        target_num = 0
    elif sort_by == "date" :
        target_num = 1
    elif sort_by == "maximum" :
        target_num = 2
    elif sort_by == "remain" :
        target_num = 3
        
    arr = [0]*len(a)
    answer = [[]]*len(a)
    
    for i in a :
        cnt = 0
        for j in a :
            if i == j :
                continue
            if i[target_num] > j[target_num] :
                cnt += 1
        answer[cnt] = i
                
    
    return answer